hack/update-deps.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
```

Comment:
I'd love to keep this repo 100% shell-free :/

Is there any alternative to adding this? @liggitt might have an idea?

Reply:
I like the approach of super-minimal makefiles (e.g. one-line target definitions, no external file includes) that just call go commands, e.g.:

make test == go test ./…
make deps == go mod tidy && go mod vendor

It technically is still shell-adjacent, but with those two rules it tends to be less of a dumping ground for crazy stuff.

Also, is it necessary to vendor?

Reply:
Not needed to vendor but I think it was originally, I can double check.

Reply:
a fake vendor directory does not work in go 1.14, it confuses go mod to only look in vendor and not the cache. I moved the fake package to testdata.